Wrote initial Privacy Considerations (#187)
openid/OpenID4VCI: openid-4-verifiable-credential-issuance-1_0.md

To sender-constrain Access Tokens, see the recommendations in Section 4.10.1 in [@!I-D.ietf-oauth-security-topics]. If Bearer Access Tokens are stored by the Wallet, they MUST be stored in a secure manner, for example, encrypted using a key stored in a protected key store.

# Privacy Considerations {#privacy-considerations}

Privacy harms can occur if information about a person is released to another party without the person's consent.

> True consent involves both the person knowing what information is being released and knowing who it is being released to, and for what purpose.

Reviewer: this sounds more relevant to presentation than issuance. it can be relevant for the issuance too, but if so, needs to be more precise.

Reviewer 2: I guess that generically applies to all kinds of flows. In issuance that would mean, a credential is issued to a wallet (or an entity posing as a wallet) not under the control of the actual user.

Author: I incorporated Torsten's feedback.

It is up to the Credential Issuer whether to update both the signature and the claim values, or only the signature.

# Privacy Considerations {#privacy-considerations}

Privacy harms can occur if information about a person is released to another party without the person's consent.
True consent involves both the person knowing what information is being released and knowing who it is being released to, and for what purpose.
The means to ensure the session's integrity to prevent disclosure to unintended parties, including wallets not under control of the End-User, are means to protect privacy.

> The Issuer needs to make choices about which claims are selectively disclosable (and should do so carefully).

Reviewer: - change this to a more generic statement: `selective disclosure should be used to protect user privacy.`

discussed that there are credential format specific aspects - atomic credentials is another option - one claim per credential. and in anoncreds, all claims become selectively disclosable without the issuer choice. But agreed to not to talk about specific credential formats in VCI here.

Claims that are always disclosed can enable correlation by verifiers.

Multiple credentials can be linkable.

> Batch issuance solves this potential privacy issue, but only when used correctly.

Reviewer: Gabe: `but only when used correctly` is not clear. -> `Batch Issuance can solve~`.

It is up to the Credential Issuer whether to update both the signature and the claim values, or only the signature.

# Privacy Considerations {#privacy-considerations}

Implementers and deployers should utilize mechanisms to protect End-User privacy.
Means of protecting End-User privacy are described below.

> Privacy harms can occur if information about a person is released to another party without the person's consent.

Reviewer 3: not only this, multiple seemingly 'innocent' consenting releases can result in privacy erosion through collusion and correlation -- there may be unintended 2nd or 3rd order effects, which is why minimal disclosure is so crucial

Author: I added this point to the section.

True consent involves both the person knowing what information is being released and knowing who it is being released to, and for what purpose.
The means to ensure the session's integrity to prevent disclosure to unintended parties, including wallets not under control of the End-User, are means to protect privacy.

> Selective disclosure should be used to protect End-User privacy, when possible.

Reviewer 3: ```suggestion
Privacy preserving techniques such as selective disclosure should be used to protect End-User privacy, when possible.
```

If supported by the Credential format, the Issuer should make choices about which claims are selectively disclosable.
Claims that are always disclosed can enable correlation by verifiers.
Another possibility is issuing credentials containing only a single claim each.

> Multiple credentials can be linkable.

Reviewer 3: ```suggestion
Multiple credentials can be linkable via claim data, identifiers, and other credential metadata.
```

> Batch issuance can solve this potential privacy issue.

Reviewer 3: would be good to add a sentence expounding on how

Privacy harms can occur if information about a person is released to another party without the person's consent.
True consent involves both the person knowing what information is being released and knowing who it is being released to, and for what purpose.
The means to ensure the session's integrity to prevent disclosure to unintended parties, including Wallets not under control of the End-User, are means to protect privacy.

Privacy-preserving mechanisms such as selective disclosure should be used to protect End-User privacy, when possible.
If supported by the Credential format, the Issuer should make choices about which claims are selectively disclosable.
Claims that are always disclosed can enable correlation by verifiers.
Another possibility is issuing Credentials containing only a single claim each.

Furthermore, multiple seemingly innocent consenting information releases can result in privacy erosion through collusion and correlation.
There may be unintended second or third order effects, which is why minimal disclosure is crucial.

Multiple Credentials can be linkable via claim data, identifiers, and other credential metadata.

> Batch issuance can solve this potential privacy issue, enabling each issued credential to only be used once.

Reviewer 4: I think this sentence probably overstates things a little as batch issuance can only help with one small part of the problem the previous sentence introduces - it does not help at all if the claim data or identifiers are linkable.

Here's a suggestion though it feels a little clumsy:

```suggestion
Using an issued credential only once can help avoid some of these concerns for some credential formats.  Batch issuance can help achieve the wallet obtain copies of a credential.
```

Author: @jogu can you change "Batch issuance can help achieve the wallet obtain copies of a credential." in your suggestion to "Batch issuance can be used by the wallet obtain multiple copies of a credential."  The current suggestion text is confusing.

Reviewer 4: I think confusing was polite, I'd clearly got halfway through rewriting the sentence and seemingly given up 😆

I've updated my suggestion with your wording (except I added a missing 'to' before 'obtain').

An Issuer normally should not learn where an End-User uses a Credential.
The Wallet should take care to not include information in the Authorization Request that may leak this information in an ad-hoc issuance scenario (e.g., a `state` value that contains clear-text session information or a `redirect_uri` that encodes this information).

There's a potential for leaking information about the Wallet to third parties when a Wallet reacts to a Credential Offer that was sent to its custom scheme.
An attacker may send such requests, wait for the Wallet to react (e.g., retrieve metadata about the "Issuer" which in reality is an attacker server), and therefore learn which Wallet is installed (e.g., by observing specific headers).
This should be avoided, e.g., by requiring user interaction with the Wallet before reacting to the offer.

> Many of the attacks described in (#security-considerations) are about unautorized parties gaining access to information.

Reviewer 5: ```suggestion
Many of the attacks described in (#security-considerations) are about unauthorized parties gaining access to information.
```
